ppl/archive/chunk/writer.go

```diff
@@ -31,30 +34,59 @@ type Writer struct {
 	wroteFirst     bool
 }
 
-func NewWriter(ctx context.Context, dir iosrc.URI, order zbuf.Order, masks []ksuid.KSUID, opts zngio.WriterOpts) (*Writer, error) {
+type Options struct {
+	Order       zbuf.Order
+	Masks       []ksuid.KSUID
+	Definitions index.Definitions
+	ZngWriter   zngio.WriterOpts
+}
+
+func NewWriter(ctx context.Context, dir iosrc.URI, opts Options) (*Writer, error) {
 	id := ksuid.New()
 	out, err := iosrc.NewWriter(ctx, dir.AppendPath(ChunkFileName(id)))
 	if err != nil {
 		return nil, err
 	}
 	counter := &writeCounter{bufwriter.New(out), 0}
-	dataFileWriter := zngio.NewWriter(counter, opts)
+	dataFileWriter := zngio.NewWriter(counter, opts.ZngWriter)
 	seekIndexPath := chunkSeekIndexPath(dir, id)
-	seekIndex, err := seekindex.NewBuilder(ctx, seekIndexPath.String(), order)
+	seekIndex, err := seekindex.NewBuilder(ctx, seekIndexPath.String(), opts.Order)
 	if err != nil {
 		return nil, err
 	}
+	idxWriter := indexWriter(nopIndexWriter{})
+	// For a Chunk writer we only care about writing index defs whose input
+	// path is empty (i.e. references the chunk data itself).
+	if l := opts.Definitions.StandardInputs(); len(l) > 0 {
+		dir := chunkZarDir(dir, id)
+		idxWriter, err = index.NewMultiWriter(ctx, dir, l)
```

Comment:
`l` is a surprising name for something with type `Definitions`.
```suggestion
	if defs := opts.Definitions.StandardInputs(); len(defs) > 0 {
		dir := chunkZarDir(dir, id)
		idxWriter, err = index.NewMultiWriter(ctx, dir, defs)
```